src/verifier/step4/SumcheckLogic.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.16;
+
+import "src/pasta/Pallas.sol";
+import "src/pasta/Vesta.sol";
+import "src/verifier/step4/KeccakTranscript.sol";
+import "src/verifier/step4/EqPolynomial.sol";
+import "src/Polynomial.sol";
+
+library PrimarySumcheck {
+    struct SumcheckProof {
```

Comment:
I think this struct can also be moved to PolyLib as soon as it deals just with uint256s internally.